### PR TITLE
Update `configure_optimizers` docs

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Fixes inconsistencies found reading the docs

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [n/a] Did you make sure to update the documentation with your changes? (if necessary)
- [n/a] Did you write any new necessary tests? (not for typos and docs)
- [n/a] Did you verify new and existing tests pass locally with your changes?
- [n/a] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

### CODE REVIEW

File: pytorch_lightning/core/lightning.py [hunk 1]
```diff
@@ -1123,6 +1123,25 @@ def configure_optimizers(self):
             - **Tuple of dictionaries** as described above, with an optional ``"frequency"`` key.
             - **None** - Fit will run without any optimizer.
 
+        Note:
```
Comment (carmocca) [inline pytorch_lightning/core/lightning.py:1126]: Moved this up to give it more visibility

File: pytorch_lightning/core/lightning.py [hunk 2]
```diff
@@ -1148,33 +1167,6 @@ def configure_optimizers(self):
...
-            Only the ``"scheduler"`` key is required, the rest will be set to the defaults above.
-
-        Note:
-            The ``"frequency"`` value is an ``int`` corresponding to the number of sequential batches optimized with the
-            specific optimizer. It should be given to none or to all of the optimizers.
-
-            There is a difference between passing multiple optimizers in a list and passing multiple optimizers in
-            dictionaries with a frequency of 1:
-            In the former case, all optimizers will operate on the given batch in each optimization step.
-            In the latter, only one optimizer will operate on the given batch at every step.
```
Comment (carmocca) [inline pytorch_lightning/core/lightning.py:1176]: Removed this block. It is duplicated

File: pytorch_lightning/core/lightning.py [hunk 3]
```diff
@@ -1226,18 +1218,6 @@ def configure_optimizers(self):
...
-
-              .. code-block:: python
-
-                  lr_dict = {
-                      'scheduler': lr_scheduler,
-                      'interval': 'step',  # or 'epoch'
-                      'monitor': 'val_f1',
-                      'frequency': x,
-                  }
-
```
Comment (carmocca) [inline pytorch_lightning/core/lightning.py:1240]: Removed this comment, we already have the explanation of the `lr_dict` at the top

**Review by ananthsub — APPROVED** [review]

File: pytorch_lightning/core/lightning.py [hunk 4]
```diff
@@ -1123,6 +1123,25 @@ def configure_optimizers(self):
...
+
+                lr_dict = {
+                    'scheduler': lr_scheduler, # The LR scheduler instance (required)
+                    # The unit of the scheduler's step size, could also be 'step'
+                    'interval': 'epoch',
+                    'frequency': 1, # The frequency of the scheduler
+                    'monitor': 'val_loss', # Metric for `ReduceLROnPlateau` to monitor
+                    'strict': True, # Whether to crash the training if `monitor` is not found
+                    'name': None, # Custom name for `LearningRateMonitor` to use
+                    'reduce_on_plateau': False, # For `ReduceLROnPlateau` scheduler, set automatically
```
Comment (carmocca) [inline pytorch_lightning/core/lightning.py:1140]: Removing this one as the user doesn't need to know about it

```suggestion
```

**Review by awaelchli — APPROVED** [review]
great. I like the truncation of the api docs here! Better to have the user guide in a separate section. <3

**Review by s-rog — APPROVED** [review]

**Review by akihironitta — APPROVED** [review]

**Review by Borda — APPROVED** [review]
